components/cdc/src/initializer.rs

```diff
@@ -224,6 +232,30 @@ impl<E: KvEngine> Initializer<E> {
             Err(box_err!("scan canceled"))
         };
 
+        let mut scanner_iter: Option<ScannerIter<S>>;
+        if kv_api == ChangeDataRequestKvApi::TiDb {
+            let scanner = ScannerBuilder::new(snap, TimeStamp::max())
+                .fill_cache(false)
+                .range(None, None)
+                .hint_min_ts(hint_min_ts)
+                .build_delta_scanner(self.checkpoint_ts, TxnExtraOp::ReadOldValue)
+                .unwrap();
+
+            scanner_iter = Some(ScannerIter::Scanner(scanner));
+        } else {
+            let mut iter_opt = IterOptions::default();
+            iter_opt.set_lower_bound(b"r", 1);
```

Comment:
Every puller from CDC request for data of only one region. So we should use KeySpan here to limit cost of scan.